armory/art_experimental/attacks/carla_obj_det_adversarial_patch.py

```diff
@@ -48,12 +49,34 @@ def create_initial_image(self, size, hsv_lower_bound, hsv_upper_bound):
         create perturbation mask based on HSV bounds
         """
         module_path = globals()["__file__"]
+        module_folder = os.path.dirname(module_path)
         # user-defined image is assumed to reside in the same location as the attack module
         patch_base_image_path = os.path.abspath(
-            os.path.join(os.path.join(module_path, "../"), self.patch_base_image)
+            os.path.join(module_folder, self.patch_base_image)
         )
+        # if the image does not exist, check cwd
+        if not os.path.exists(patch_base_image_path):
+            patch_base_image_path = os.path.abspath(
+                os.path.join(paths.runtime_paths().cwd, self.patch_base_image)
+            )
+        # image not in cwd or module, check if it is a url to an image
+        if not os.path.exists(patch_base_image_path):
+            import requests
+
+            try:
+                response = requests.get(self.patch_base_image, allow_redirects=True)
```

Comment:
Any worry about safety of links? i.e. zipbombs or something?

Reply:
cf. https://stackoverflow.com/questions/14270698/get-file-size-using-python-requests-while-only-getting-the-header for sample usage

Reply:
Resolved with https://github.com/twosixlabs/armory/pull/1939/commits/67e002049b4c5c5a03f7b1074231c5346da52ae7